### DISCUSSION TITLE
GetAll Method from the ModelServerClient.java does not return our models

### DISCUSSION THREAD

**Discussion by NomsaT** [eclipse-emfcloud/emfcloud#113] [ModelServer] [answered]
Currently trying to access all the models from our workspace using java API but the getAll method returns an empty list + some warnings in the console indicating that the models are incomplete. By calling the API directly from Postman, it retrieves all models without problem.
Does anyone know if our models need to be in a specific format for them to be "complete" because from our observations the two conditions for the if statement inside the getAll method are false( modelUri.isPresent() && content.isPresent()) and we are not sure why this is the case.

**Comment by vhemery** [comment, marked as answer]
> By calling the API directly from Postman, it retrieves all models without problem.

Can you post the result from the API call with Postman ? This would make it easier to analyze your problem. (check whether it comes from the java API layer only or from the backend and the sent data)

**Reply by vhemery** [reply]
My guess is that your reply would look like 
```
{
   './myModelUri' : {
      ...
   },
   './myOtherModelUri' : {
    ...
   }
}
```
which is a correct json map.
It looks like the ModelServerClient, though, is expecting a json node with two fields named 'modeluri' and 'content'...
https://github.com/eclipse-emfcloud/emfcloud-modelserver/blob/0031a4452de4907d4dac548de9d1e96934f4448e/bundles/org.eclipse.emfcloud.modelserver.client/src/org/eclipse/emfcloud/modelserver/client/ModelServerClient.java#L155

If you can confirm this, you can submit a bug at ... well, looking for the url, I see that you already submitted https://github.com/eclipse-emfcloud/emfcloud-modelserver/issues/156
So this discussion will be referenced there...

**Reply by NomsaT** [reply]
Yes your guess was correct

on the API call with PostMan we get something like this:
{
    "type": "success",
    "data": {
        './myModelUri' : {
      ...
       },
      './myOtherModelUri' : {
    ...
     }
}
and then unfortunately on the Java API call we get the following:
18347 [main] WARN ModelServerClient  - Incomplete Model: {"eClass":"http://www.eclipse.org/emf/2002.......}
18347 [main] WARN ModelServerClient  - Incomplete Model: {eClass":"http://www.eclipse.org/emf/2002/Ecore#//EPackage","eAnnotations":............}
18348 [main] WARN ModelServerClient  - Incomplete Model: {"eClass":"http://iso.org........}

**Reply by martin-fleck-at** [reply]
@NomsaT Thank you very much for reporting that. It seems that there was some update that did not make it into the Java Client API, it works fine when using the Node API. Thanks to @vhemery I was able to create a fix quickly: https://github.com/eclipse-emfcloud/emfcloud-modelserver/pull/164. As soon as this is merged, it should be available in the nightly update site. If you need it earlier you can always subclass the client and apply the fix yourself.

If you feel everything has been answered, please mark the answer so this discussion can be closed.
